include/exadg/incompressible_navier_stokes/spatial_discretization/operator_coupled.h

```diff
@@ -68,13 +68,13 @@ class NonlinearOperatorCoupled
    * 'evaluate_residual'.
    */
   void
-  evaluate_residual(BlockVectorType & dst, BlockVectorType const & src) const
+  evaluate_residual(BlockVectorType & dst, BlockVectorType const & src)
   {
     pde_operator->evaluate_nonlinear_residual(dst, src, rhs_vector, time, scaling_factor_mass);
   }
 
 private:
-  PDEOperator const * pde_operator;
+  PDEOperator * pde_operator;
```

Comment:
I don't think just removing the `const` ness of the objects and functions is generally a good idea. If I am not mistaken, this all boils down to the `update_viscosity()` call down the line. None of these operators and functions are actually supposed to modify the `PDEOperator` object, just manage it. In the non-linear case, the viscosity model seems to be an exception that we have to update. In my opinion, making it `mutable` in `SpatialOperatorBase` will make it clearer that the viscosity model is subject to change, but everthing else cannot be changed. This should allow you to keep the `const` qualifier everywhere.